## Converting nyc_geo.json into dataframe

In [5]:
# required libraries
import json
import pandas as pd

In [8]:
# open and load json
nycd = open('nyc_geo.json')
nyc_geo = json.load(nycd)

In [11]:
# check json keys
nyc_geo.keys()

dict_keys(['type', 'totalFeatures', 'features', 'crs', 'bbox'])

After looking at json file, it appears all we really need from here is name, longitude, and latitude of each neighborhood

In [ ]:
# longitude is 0, latitude is 1

In [44]:
# create longitude list to append to dictionary
long_list = []
for i in nyc_geo['features']:
    long_list.append(i['geometry']['coordinates'][0])

In [45]:
# create latitude list to append to dictionary
lat_list = []
for i in nyc_geo['features']:
    lat_list.append(i['geometry']['coordinates'][1])

In [46]:
# create name list to append to dictionary
name_list = []
for i in nyc_geo['features']:
    name_list.append(i['properties']['name'])

In [47]:
# instantiate empty dictionary
nyc_dict = {}

In [49]:
# append relevant data to dictionary
nyc_dict['name'] = name_list
nyc_dict['longitude'] = long_list
nyc_dict['latitude'] = lat_list

In [55]:
# convert dictionary into dataframe
df_nyc = pd.DataFrame.from_dict(nyc_dict)

In [61]:
df_nyc

name  longitude   latitude
0       Wakefield -73.847201  40.894705
1      Co-op City -73.829939  40.874294
2     Eastchester -73.827806  40.887556
3       Fieldston -73.905643  40.895437
4       Riverdale -73.912585  40.890834
..            ...        ...        ...
301  Hudson Yards -74.000111  40.756658
302       Hammels -73.805530  40.587338
303     Bayswater -73.765968  40.611322
304  Queensbridge -73.945631  40.756091
305     Fox Hills -74.081740  40.617311

[306 rows x 3 columns]

Each neighborhood will be used as a sample in our unsupervised clustering. (Total of 306 samples)

In [58]:
# save dataframe as csv on local machine
df_nyc.to_csv('nyc_geo_dataframe.csv', index=False)

In [63]:
df_nyc.loc[50]

name         Gravesend
longitude     -73.9735
latitude       40.5953
Name: 50, dtype: object